spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -101,9 +92,10 @@ public class RewriteDataFilesSparkAction
   private boolean partialProgressEnabled;
   private boolean removeDanglingDeletes;
   private boolean useStartingSequenceNumber;
-  private RewriteJobOrder rewriteJobOrder;
-  private FileRewriter<FileScanTask, DataFile> rewriter = null;
   private boolean caseSensitive;
+  private BinPackRewriteFilePlanner planner = null;
+  private FileRewriteRunner<FileGroupInfo, FileScanTask, DataFile, RewriteFileGroup> runner = null;
+  private boolean shufflingPlanner = false;
```

Comment:
Why a separate variable here? Can't we just check "if Rewriter extends SortRewriter"

Reply:
Good idea.
Done